internal/consensus/reactor.go

```diff
@@ -292,6 +293,95 @@ conR:
 	}
 }
 
+// UpdateMetrics the applied block from the block pool updates the state metrics during the fast-sync.
+// Some implementation details please check State.recordMetrics.
+func (r *Reactor) UpdateMetrics(s sm.State, b *types.Block) {
+	rm := r.Metrics
+
+	rm.Validators.Set(float64(s.Validators.Size()))
+	rm.ValidatorsPower.Set(float64(s.Validators.TotalVotingPower()))
+
+	var (
+		missingValidators      int
+		missingValidatorsPower int64
+	)
+
+	if b.Height > s.InitialHeight {
+		var (
+			commitSize = b.LastCommit.Size()
+			valSetLen  = len(s.LastValidators.Validators)
+			address    types.Address
+		)
+		if commitSize != valSetLen {
+			panic(fmt.Sprintf("commit size (%d) doesn't match valset length (%d) at height %d\n\n%v\n\n%v",
+				commitSize, valSetLen, b.Height, b.LastCommit.Signatures, s.LastValidators.Validators))
+		}
+
+		if rs := r.state; rs != nil {
+			if rs.privValidator != nil {
+				if rs.privValidatorPubKey == nil {
+					rs.Logger.Error(fmt.Sprintf("recordMetrics: %v", errPubKeyIsNotSet))
+				} else {
+					address = rs.privValidatorPubKey.Address()
+				}
+			}
+			for i, val := range rs.LastValidators.Validators {
+				commitSig := b.LastCommit.Signatures[i]
+				if commitSig.Absent() {
+					missingValidators++
+					missingValidatorsPower += val.VotingPower
+				}
+
+				if bytes.Equal(val.Address, address) {
+					label := []string{
+						"validator_address", val.Address.String(),
+					}
+					rm.ValidatorPower.With(label...).Set(float64(val.VotingPower))
+					if commitSig.ForBlock() {
+						rm.ValidatorLastSignedHeight.With(label...).Set(float64(b.Height))
+					} else {
+						rm.ValidatorMissedBlocks.With(label...).Add(float64(1))
+					}
+				}
+
+			}
+		}
+
+	}
+
+	rm.MissingValidators.Set(float64(missingValidators))
+	rm.MissingValidatorsPower.Set(float64(missingValidatorsPower))
+
+	var (
+		byzantineValidatorsPower = int64(0)
+		byzantineValidatorsCount = int64(0)
```

Comment:
I think you can just do

```
var value int64
```
the 0 value is implicit

Reply:
Sure, thanks.